crates/bevy_pbr/src/render/pbr.wgsl

```diff
@@ -138,7 +138,7 @@ fn fragment(
         pbr_input.V = V;
         pbr_input.occlusion = occlusion;
 
-        pbr_input.flags = mesh.flags;
+        pbr_input.flags = mesh[in.instance_index].flags;
```

Comment:
this is using `in.instance_index` unconditionally, but it's existence depends on `VERTEX_OUTPUT_INSTANCE_INDEX`. i think that's probably fine since this frag shader is (currently) specific to standard materials, but it doesn't feel great.

as far as i can tell this is the only use of the instance index in any fragment stage. would it make more sense to push the mesh flags through the `MeshVertexOutput` struct directly (and unconditionally)? then we could get rid of the `VERTEX_OUTPUT_INSTANCE_INDEX` entirely.

this would not make sense if there are other potential uses for mesh data in the fragment stage but i think there shoudn't be generally.